lib/matplotlib/dates.py

```diff
@@ -212,47 +212,31 @@ def _to_ordinalf(dt):
     days, preserving hours, minutes, seconds and microseconds.  Return value
     is a :func:`float`.
     """
-
-    if hasattr(dt, 'tzinfo') and dt.tzinfo is not None:
-        delta = dt.tzinfo.utcoffset(dt)
-        if delta is not None:
-            dt -= delta
+    # Convert to UTC
+    tzi = getattr(dt, 'tzinfo', None)
+    if tzi is not None:
+        dt = dt.astimezone(UTC)
+        tzi = UTC
 
     base = float(dt.toordinal())
-    if isinstance(dt, datetime.datetime):
-        # Get a datetime object at midnight in the same time zone as dt.
-        cdate = dt.date()
-        midnight_time = datetime.time(0, 0, 0, tzinfo=dt.tzinfo)
+
+    # If it's sufficiently datetime-like, it will have a `date()` method
+    cdate = getattr(dt, 'date', lambda: None)()
+    if cdate is not None:
```

Comment:
it is calling the attribute it obtains before assigning to cdate.

On Mon, May 16, 2016 at 3:29 PM, Eric Firing <notifications@github.com>
wrote:

> In lib/matplotlib/dates.py
> <https://github.com/matplotlib/matplotlib/pull/6262#discussion_r63410423>:
>
> >
> >      base = float(dt.toordinal())
> > -    if isinstance(dt, datetime.datetime):
> > -        # Get a datetime object at midnight in the same time zone as dt.
> > -        cdate = dt.date()
> > -        midnight_time = datetime.time(0, 0, 0, tzinfo=dt.tzinfo)
> > +
> > +    # If it's sufficiently datetime-like, it will have a `date()` method
> > +    cdate = getattr(dt, 'date', lambda: None)()
> > +    if cdate is not None:
>
> Based on the line above, I don't think cdate will ever be None--it can
> *return* None.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/6262/files/3e6e8d1324e4aac4c0cf08dcffab451e1ee5f525#r63410423>
>